app/views/admin/shelters/index.html.erb

```diff
@@ -0,0 +1,21 @@
+<h1>All Shelters</h1>
+
+<%= link_to "Sort by number of pets", "/shelters?sort=pet_count" %>
+
+<%= form_with url: "/shelters", method: :get, local: true do |f| %>
+  <%= f.label :search %>
+  <%= f.text_field :search %>
+  <%= f.submit "Search" %>
+<% end %>
+
+<% @shelters.each do |shelter| %>
+  <div id="shelter-<%= shelter.id %>">
+    <h3><%= shelter.name %></h3>
+    <p>Created at: <%= shelter.created_at %></p>
+    <%= link_to "Update #{shelter.name}", "/shelters/#{shelter.id}/edit" %>
+    <%= link_to "Delete #{shelter.name}", "/shelters/#{shelter.id}", method: :delete %>
+  </div>
+<% end %>
+
+<h2>Shelters With Pending Applications</h2>
+<%= @shelters.pending_apps.uniq.pluck(:name) %>
```

Comment:
Calling `uniq` on a collection is a very expensive operation. Imagine you have 1,000,000 records and there are duplicates, the way you have it now, your application will retrieve all 1,000,000 records and then work through all of them to filter out the duplicates. 

A better way would be to not return duplicates in the first place. Because a database is optimized to remove duplicate records with a DISTINCT clause, we want to perform this operation on the database, thus limiting the number of entries that we pull into our application.